api.go

```diff
@@ -330,6 +330,11 @@ func (client *Client) chooseLUN(initiatorName string) (int, error) {
 
 	sort.Sort(Volumes(volumes))
 
+	klog.V(5).Infof("checking if next LUN is not above maximum LUNs limit")
+	if volumes[len(volumes)-1].LUN+1 < MaximumLUN {
```

Comment:
Looks like this could throw an error for a negative slice index if len(volumes) is 0, checked in the next block

Reply:
Yes, good catch. Corrected to handle a volume slice of zero length.